### ISSUE TITLE
downloadHandler seems can't recognize Chinese filename

### ISSUE THREAD

**Issue by shizidushu** [rstudio/shiny#1858]
library(shiny)
    
    ui <- fluidPage(
       
       titlePanel("Can I download a file with chinese filename?"),
       
       downloadButton("downloadData", "Download")
    )
    
    
    server <- function(input, output) {
      # Our dataset
      data <- mtcars
      
      output$downloadData <- downloadHandler(
        filename = function() {
          paste("中文data-", Sys.Date(), ".csv", sep="")
        },
        content = function(file) {
          write.csv(data, file)
        }
      )
    }
    
    # Run the application 
    shinyApp(ui = ui, server = server)
    
When I click the downbutton, the downloaded filename is '%FFFFFFE4%FFFFFFB8%FFFFFFAD%FFFFFFE6%FFFFFF96%FFFFFF87data-2017-09-28.csv'

**Comment by wch** [comment]
The problem seems to be that shiny's internal version of `URLencode()` does something different from what it should

```R
shiny:::URLencode("中文data-2017-09-28.csv")
#> [1] "%FFFFFFE4%FFFFFFB8%FFFFFFAD%FFFFFFE6%FFFFFF96%FFFFFF87data-2017-09-28.csv"

# This gives the correct result
utils::URLencode("中文data-2017-09-28.csv")
#> [1] "%E4%B8%AD%E6%96%87data-2017-09-28.csv"
```

It's called from these places for downloads:
https://github.com/rstudio/shiny/blob/176fe69/R/shiny.R#L1687
https://github.com/rstudio/shiny/blob/176fe69/R/shiny.R#L1775

**Comment by wch** [comment]
The root cause of this is rstudio/httpuv#86.

**Comment by wch** [comment]
This has been fixed by rstudio/httpuv#87.
